client/engine/chainservice/mockchain.go

```diff
@@ -16,7 +16,7 @@ type MockChain struct {
 
 	transListener chan protocols.ChainTransaction   // this is used to broadcast transactions that have been received
 	holdings      map[types.Destination]types.Funds // holdings tracks funds for each channel
-	blockNum      *uint64
+	blockNum      *uint64                           // MockChain is often passed around by value. The pointer allows for shared state.
```

Comment:
I've looked into passing the mock chain by reference here #731 .

Reply:
Gotcha. My goal here is to make it harder to create unexpected behavior if MockChain is passed by value. When MockChain is passed by value and the MockChain struct has primitive (as opposed to pointer) fields, it's possible to run into deadlock since MockChain values will not share the block number.